### ISSUE TITLE
`ScopeGuard` should be called `InitializationScopeGuard` or something

### ISSUE THREAD

**Issue by dhollman** [kokkos/kokkos#2059]
`ScopeGuard` doesn't say what it's guarding (like, e.g., `std::lock_guard` does).

**Comment by dalg24** [comment]
How about `ExecEnvGuard` since it initializes and finalizes the execution environment?

**Comment by mhoemmen** [comment]
Name kinda makes sense to me -- it's guarding "the scope in which you're allowed to use Kokkos" -- but I'm not invested in the name.

**Comment by dhollman** [comment]
@mhoemmen The problem is that `ScopeGuard` is far more general than just for Kokkos. Scope guards have been around since the 90s.  There is a proposal almost as old as `mdspan` to add a general scope guard to the C++ standard library ([P0052](http://wg21.link/p0052)). The scope guard in that proposal is something much more general than the thing with that name in Kokkos.  I actually thing that Kokkos (or probably `stdoe`, once that gets rolling) should definitely provide a general scope guard for our users, but the thing named `ScopeGuard` right now is definitely not it.

**Comment by mhoemmen** [comment]
@dhollman True, P0052 claims that name for something super general.  I'm not attached to the name but don't have better ideas at the moment :-)
